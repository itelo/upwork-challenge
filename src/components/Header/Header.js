import React from "react";
import PropTypes from "prop-types";
import { withRouter } from "react-router-dom";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Grid from "@material-ui/core/Grid";
import Typography from "@material-ui/core/Typography";
import withStyles from "@material-ui/core/styles/withStyles";
import IconButton from "@material-ui/core/IconButton";
import ArrowBackIcon from "@material-ui/icons/ArrowBack";

// Header Styles to class
const styles = {
  // set color of navigation bar
  headerRoot: {
    backgroundColor: "#008394",
    position: "relative"
  },
  // position of back button
  backButton: {
    position: "absolute",
    top: 4,
    left: 4
  }
};

// location.state contains the index of selected drink
// it is used to scroll to that drink in list

const Header = ({ classes, location, history }) => (
  <AppBar classes={{ root: classes.headerRoot }} position="absolute">
    <Toolbar>
      {location.pathname !== "/" && (
        <IconButton
          className={classes.backButton}
          color="inherit"
          aria-label="ArrowBack"
          onClick={() => history.push("/", location.state)}
        >
          <ArrowBackIcon />
        </IconButton>
      )}
      <Grid container direction="row" justify="center" alignItems="center">
        <Grid item>
          <Typography variant="title" color="inherit">
            Random Drinks 0.1
          </Typography>
        </Grid>
      </Grid>
    </Toolbar>
  </AppBar>
);

Header.propTypes = {
  classes: PropTypes.shape({
    headerRoot: PropTypes.string.isRequired,
    backButton: PropTypes.string.isRequired
  }).isRequired,
  location: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired
};

export default withStyles(styles)(withRouter(Header));
