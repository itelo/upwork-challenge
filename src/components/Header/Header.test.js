import React from "react";
import { MemoryRouter } from "react-router";
import Header from "./Header";
import { configure } from "enzyme";
import { createMount, createShallow } from "@material-ui/core/test-utils";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("<Header />", () => {
  const mount = createMount();
  const shallow = createShallow();
  const drinkPath = "/drinks/1";
  const home = "/";
  it("test icon button in / to be 0", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={[home]}>
        <Header />
      </MemoryRouter>
    );
    expect(wrapper.find("IconButton").length).toBe(0);
  });

  it("test icon button in /drinks/1 to be 1", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={[drinkPath]}>
        <Header />
      </MemoryRouter>
    );
    expect(wrapper.find("IconButton").length).toBe(1);
  });

  it("test icon button func to go from /drinks/1 to /", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={[drinkPath]}>
        <Header />
      </MemoryRouter>
    );
    expect(wrapper.find("Router").props().history.location.pathname).toBe(
      drinkPath
    );
    // history push to  "/"
    wrapper
      .find("IconButton")
      .props()
      .onClick();
    expect(wrapper.find("Router").props().history.location.pathname).toBe(
      home
    );
    expect(wrapper.find("IconButton").length).toBe(1);
  });

  it("snapshot test /drinks", () => {
    let wrapper = shallow(
      <MemoryRouter initialEntries={[drinkPath]}>
        <Header />
      </MemoryRouter>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
