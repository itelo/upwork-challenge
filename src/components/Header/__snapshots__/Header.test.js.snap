// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> snapshot test /drinks 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MemoryRouter
    initialEntries={
        Array [
            "/drinks/1",
          ]
    }
>
    <WithStyles(withRouter(Header)) />
</MemoryRouter>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithStyles(withRouter(Header)) />,
      "history": Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "hv6ixf",
            "pathname": "/drinks/1",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "hv6ixf",
          "pathname": "/drinks/1",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(withRouter(Header)) />,
        "history": Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "hv6ixf",
              "pathname": "/drinks/1",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "hv6ixf",
            "pathname": "/drinks/1",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "context": Object {},
    "hydrateIn": undefined,
  },
}
`;
